lib/loggers/index.js

```diff
@@ -51,31 +52,42 @@ const container = new winston.Container();
 // override get to use a wrapper so loggers that are retrieved via `.get()`
 // prior to logger initialization will receive the updated configuration
 // when logging after initialization
-container._get = container.get;
+// container._get = container.get;
 container._add = container.add;
-container.get = container.add = function(id) {
-  const existing = container.loggers[id];
-  let logger = container._get.apply(container, arguments);
+container.add = function(id) {
+  const existing = container.has(id);
+  let logger = container._add.apply(container, arguments);
   if(!existing) {
     const wrapper = Object.create(logger);
-    wrapper.log = function(level, msg, meta) {
-      // merge per-logger and per-log meta and call parent logger
-      meta = Object.assign({}, this.meta, meta);
-      return Object.getPrototypeOf(wrapper).log.apply(
-        wrapper, [level, msg, meta]);
+    wrapper.log = function(level, msg, ...meta) {
+      // get default meta to ensure it is always applied
+      const {defaultMeta} = this;
+      // use `Object.getPrototypeOf` because the prototype may change
+      // while the logging subsystem initializes
+      const log = Object.getPrototypeOf(wrapper).log;
+      if(arguments.length === 1) {
+        return log.apply(wrapper, [{...defaultMeta, ...level}]);
```

Comment:
@dlongley `defaultMeta` is added in winston here: https://github.com/winstonjs/winston/blob/master/lib/winston/create-logger.js#L80

You comfortable with simplifying this in the single argument case?

Reply:
Yeah, I'm fine if it works without it. I'd rather reduce changes to the lib as much as possible.